enhancements/microshift/advertise-version-details.md

```diff
@@ -0,0 +1,302 @@
+drawbacks. First, not all console users have cluster admin privileges,
+which means that the console cannot show them the cluster version
+details. Second, the ClusterVersion API is not present in a MicroShift
+deployment, so instead [we created a
+ConfigMap](https://github.com/openshift/microshift/pull/776) in the
+`kube-public` namespace containing some basic version
+details. That means a caller must know two different APIs to check the
+version and product they are communicating with. This enhancement
+describes how we can standardize on one to make it easier for all
+authenticated users to obtain basic version details for a cluster.
+
+### User Stories
+
+As a console user, I want to see the version of OCP I am using so that
+I can refer to the correct version of the documentation and API
+references.
+
+As a console developer, I want to be able to obtain the version of OCP
+that I am using so that the console app can use different
+implementation choices for behaviors that cannot be discovered by
+probing the Kubernetes API versions supported.
+
+As an authenticated API caller, I want to obtain the product name and
+version of the OpenShift variant I am communicating with so that I can
+use it to make decisions on the client side.
+
+As an application developer, I want to be able to determine the
+Kubernetes distribution the application is being installed into at
+runtime so that I only need to implement one set of installation
+logic.
+
+As an OpenShift user without admin privileges, I want to find
+information about the version of the product I am using in a
+consistent way, regardless of whether I am using standalone OpenShift,
+HyperShift, or MicroShift.
+
+### Goals
+
+* Use a standard Kubernetes type, to keep access easy for arbitrary
+  API clients. This also lets us avoid introducing a new CRD, so we do
+  not increase the overhead for MicroShift.
+* Define a standard location for version information that all
+  OpenShift variants can support.
+
+### Non-Goals
+
+* This change does not introduce a new way to update the version of a
+  cluster.
+* This change does not introduce a new way for a client to discover
+  that a cluster is being upgraded.
+* This change does not introduce a new way for a client to discover
+  the cluster topology (single-node, HA, external control-plane, etc.).
+
+## Proposal
+
+Because we want arbitrary clients to be able to read and understand
+the data, we will use a standard Kubernetes ConfigMap.
+
+Because we want all OpenShift variants to publish the information in
+the same way and that the data should not be writable by API callers,
+we will specify the name and namespace for the ConfigMap.
+
+Because MicroShift exposes a limited number of namespaces by default,
+we need to choose from one of those. The most suitable available
+namespace is `openshift`, so we will create the ConfigMap with name
+`version` in the `openshift` namespace.
```

Comment:
Another reason for picking `openshift` namespace is that in all OCP clusters it's widely readable namespace by design, which is currently primed with such data as ImageStreams, Templates, and is also used to present MOTD for oc.

Reply:
I added this to the doc